wandb/lib/lib.py

```diff
@@ -0,0 +1,70 @@
+import ctypes
+import pathlib
+import platform
+from typing import Optional
+
+
+class Lib:
+    def __init__(self) -> None:
+        self._lib: Optional[ctypes.CDLL] = None
+        self.init_sentry()
+
+    @property
```

Comment:
consider `functools.cached_property` instead https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
Added in py3.8 :( Will replace once we drop 3.7.